homeassistant/components/hunterdouglas_powerview/config_flow.py

```diff
@@ -35,42 +36,60 @@ async def validate_input(hass: core.HomeAssistant, hub_address: str) -> dict[str
 
     try:
         async with asyncio.timeout(10):
-            device_info = await async_get_device_info(pv_request, hub_address)
+            hub = Hub(pv_request)
+            await hub.query_firmware()
+            device_info = await async_get_device_info(hub)
     except HUB_EXCEPTIONS as err:
+        _LOGGER.debug(err)
         raise CannotConnect from err
 
+    _LOGGER.debug("Connection made using api version: %s", hub.api_version)
+
     # Return info that you want to store in the config entry.
     return {
         "title": device_info.name,
         "unique_id": device_info.serial_number,
+        CONF_API_VERSION: hub.api_version,
     }
 
 
-class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+class PowerviewConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Hunter Douglas PowerView."""
 
     VERSION = 1
 
-    def __init__(self):
+    def __init__(self) -> None:
         """Initialize the powerview config flow."""
-        self.powerview_config = {}
-        self.discovered_ip = None
-        self.discovered_name = None
+        self.powerview_config: dict = {}
+        self.discovered_ip: str | None = None
+        self.discovered_name: str | None = None
+        self.data_schema: dict = {vol.Required(CONF_HOST): str}
 
-    async def async_step_user(self, user_input=None):
+    async def async_step_user(self, user_input=None) -> FlowResult:
```

Comment:
Missing typing on user_input.

I suggest merging all the typing fixes to `dev` first in another PR, which will make this smaller